frontend/src/hooks/useAddRedactionToState.jsx

```diff
@@ -0,0 +1,39 @@
+import {
+    redactionIDsState,
+    redactionsState,
+  } from "@/util/CommentRedactionState";
+  
+  import { useRecoilCallback } from "recoil";
+  
+  //call this to add a new redaction
+  export default function useAddRedactionToState() {
+    console.log(redactionIDsState)
+    return useRecoilCallback(
+      ({ set }) => (id, threadData) => {
+        set(redactionIDsState, (ids) => new Set([...Array.from(ids), id]));
+        set(redactionsState(id), threadData);
+      },
+      []
+    );
+  }
+
+  export default function useRemoveRedactionFromState(mark) {
+    return useRecoilCallback(
+      ({ set }) => (id) => {
+        console.log('Removing redaction with ID:', id);
+        set(redactionIDsState, (ids) => {
+          console.log('Current redaction IDs:', ids);
+          const filteredIds = Array.from(ids).filter(item => item !== id);
```

Comment:
what is this supposed to be filtering? you're filtering out all items that aren't `id` then creating a set from the result? isn't that the same as
```js
return id in ids ? new Set(id) : new Set()
```